src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -544,15 +554,17 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
 
             internal static bool IsSyntaxTargetForGeneration(SyntaxNode node) => node is ClassDeclarationSyntax { AttributeLists: { Count: > 0 }, BaseList: { Types : {Count : > 0 } } };
 
-            internal static ClassDeclarationSyntax? GetSemanticTargetForGeneration(GeneratorSyntaxContext context)
+            internal static ClassDeclarationSyntax? GetSemanticTargetForGeneration(GeneratorSyntaxContext context, CancellationToken cancellationToken)
             {
                 var classDeclarationSyntax = (ClassDeclarationSyntax)context.Node;
 
                 foreach (AttributeListSyntax attributeListSyntax in classDeclarationSyntax.AttributeLists)
                 {
                     foreach (AttributeSyntax attributeSyntax in attributeListSyntax.Attributes)
                     {
-                        IMethodSymbol attributeSymbol = context.SemanticModel.GetSymbolInfo(attributeSyntax).Symbol as IMethodSymbol;
+                        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Is this one a bit "too often"? We are also passing it into `GetSymbolInfo`, which should be checking for cancellation...

Reply:
there is no guarantee those calls will actually check.  on the IDE side we generally check at entrypoint and in loops.  it hasn't been a problem for us.